std/recursion/plonk/opts.go

```diff
@@ -72,3 +72,34 @@ func GetNativeVerifierOptions(outer, field *big.Int) backend.VerifierOption {
 		return nil
 	}
 }
+
+type verifierCfg struct {
+	withUseSafe bool
+}
+
+// VerifierOption allows to modify the behaviour of PLONK verifier.
+type VerifierOption func(cfg *verifierCfg) error
+
+// WithUseSafe forces the usage of safe formulas for point addition and
+// multi-scalar multiplication. The option is necessary when recursing simple
+// inner circuits whose selector polynomials may have exceptional cases (zeros,
+// equal to each other, inverses of each other).
+//
+// Safe formulas are less efficient to use, so using this option has performance
+// impact on the outer circuit size.
+func WithUseSafe() VerifierOption {
```

Comment:
I prefer `WithSafeFormulas` or `WithSafeArithmetic` or `WithCompleteFormulas` -- 

doc wise; as a circuit developer, how do I know if my inner circuit have selector polynomial with exceptional cases? 
Shouldn't we put the "safe arithmetic formulas" as default, and have an option that disable it when we know what we are doing and don't expect such edge cases??

Reply:
Yup, we thought about with @yelhousni and it seems that doing unsafe by default was better. The exceptional cases happen when:
* Qk = 0
* Ql = Qr
* Ql = -Qr
* (a few more related to relations of the selector polynomial commitments)
* batchedproofs claimed values are 0-s

These cases usually happen quite rarely when we generate dummy internal circuits for testing or when the proofs are malformed. The option is rather for the cases when we test to avoid needing to patch gnark.

cc: @yelhousni, @ThomasPiellard to confirm my intuition?

Reply:
re renaming the option - indeed, I'm not very good at naming things. I'll rename.

Reply:
right edge-cases should be trivial and I prefer to have the unsafe path as default because some of the decisions are only related to Ethereum (e.g. `(0,0)` is not the infinity point normally). Also making the safe default adds more constraints to the unsafe path, at least in the way we're are handling it right now.

Reply:
Renamed 👍